c10/core/impl/CUDATrace.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <c10/core/impl/PyInterpreter.h>
+
+namespace c10 {
+namespace impl {
+
+struct C10_API CUDATrace {
+  // When PyTorch migrates to C++20+, this should be changed to an atomic flag.
+  static std::atomic<const PyInterpreter*> cudaTraceState;
```

Comment:
Would be nice to have a little comment saying that because we are usually x86 the load/store are not locked. But because this atomic is a write-once atomic, a relaxed memory order is possible. For a write once atomic variable, where it is acceptable to race between the null and the initialized state, I think `memory_order_release` is acceptable for the `store` and a `memory_order_acquire` for the `load`.

Reply:
in fact, release/acquire should be good enough to deal with subsequent modifications of this variable, assuming the PyInterpreter never dies (which it won't, we disarm PyInterpreters but leak them when their interpreters go away, IIRC)

Reply:
Thank you for the information, I've added the memory orders like you suggested.